### PR TITLE
Add vscode formula

### PR DESCRIPTION
**Pull request by cweagans**

### CODE REVIEW

**Comment by zchee** [comment]
I think version to 0.1.0 rather than preview.
screenshot 2015-04-30 02 59 11

File: Casks/vscode.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cask :v1 => 'vscode' do
+  version 'preview'
+  sha256 '45f560cfdd2a3108ed23ec5ecff461313267ca12082dafffd351252db751c357'
+
+  url "http://download.microsoft.com/download/0/D/5/0D57186C-834B-463A-AECB-BC55A8E466AE/VSCode-osx.zip"
```
Comment (vitorgalvao) [inline Casks/vscode.rb]: Double quotes are reserved for cases where string interpolation is needed. Please change these to single quotes.

File: Casks/vscode.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  name 'Visual Studio Code'
+  homepage 'https://code.visualstudio.com/Download'
+  license :free
+
+  app 'Visual Studio Code.app'
+
```
Comment (vitorgalvao) [inline Casks/vscode.rb]: Tis line must go.

File: Casks/vscode.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+cask :v1 => 'vscode' do
+  version 'preview'
+  sha256 '45f560cfdd2a3108ed23ec5ecff461313267ca12082dafffd351252db751c357'
+
+  url "http://download.microsoft.com/download/0/D/5/0D57186C-834B-463A-AECB-BC55A8E466AE/VSCode-osx.zip"
+  name 'Visual Studio Code'
+  homepage 'https://code.visualstudio.com/Download'
+  license :free
```
Comment (elyscape) [inline Casks/vscode.rb]: Should be `:gratis`.

**Comment by elyscape** [comment]
Might be a good idea to add a `:vendor` tag:
```ruby
tags :vendor => 'Microsoft'
```

**Comment by elyscape** [comment]
You should include a `zap` stanza:
```ruby
zap :delete => [
                '~/Library/Application Support/Code',
                '~/Library/Application Support/com.github.atom-shell.ShipIt',
                '~/Library/Caches/Code',
                '~/Library/Preferences/com.github.atom-shell.plist',
               ],
```

File: Casks/vscode.rb (hunk 3 above)
Comment (vitorgalvao) [inline Casks/vscode.rb]: Nice catch, @elseym. Thank you.

Comment (elseym) [inline Casks/vscode.rb]: it wasn't me
@elyscape is the one you want

Comment (vitorgalvao) [inline Casks/vscode.rb]: @elseym Oops. You are correct.

File: Casks/vscode.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+cask :v1 => 'vscode' do
```
Comment (vitorgalvao) [inline Casks/vscode.rb]: Should be `visual-studio-code`. See the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_token_reference.md).

**Comment by vitorgalvao** [comment]
Closing in favour of the more complete #10916.

**Pull request closed without merging**